htdocs/core/class/commonobject.class.php

```diff
@@ -3715,16 +3715,25 @@ function formAddObjectLine($dateSelector, $seller, $buyer)
 
 		// Output template part (modules that overwrite templates must declare this into descriptor)
 		// Use global variables + $dateSelector + $seller and $buyer
-		$dirtpls=array_merge($conf->modules_parts['tpl'],array('/core/tpl'));
-		foreach($dirtpls as $reldir)
+		$dirtpls=array_merge($conf->modules_parts['tpl'], array('/core/tpl'));
+		foreach($dirtpls as $module => $reldir)
 		{
-			$tpl = dol_buildpath($reldir.'/objectline_create.tpl.php');
+			if (!empty($module) && empty($conf->$module->enabled)) continue;
+
+			if (!empty($module))
+			{
+				$tpl = dol_buildpath($reldir.'/objectline_create.tpl.php');
+			}
+			else
+			{
+				$tpl = DOL_DOCUMENT_ROOT.$reldir.'/objectline_create.tpl.php';
+			}
 			if (empty($conf->file->strict_mode)) {
 				$res=@include $tpl;
 			} else {
 				$res=include $tpl; // for debug
 			}
-			if ($res) break;
+			if ($res > 0) break;
```

Comment:
$res is result of include so is a boolean.
Comparing a boolean to 0 seems wrong.

Reply:
php include function can also return a value if specified in the include file : https://www.php.net/manual/en/function.include.php
"Handling Returns: include returns FALSE on failure and raises a warning. Successful includes, unless overridden by the included file, return 1. It is possible to execute a return statement inside an included file in order to terminate processing in that file and return to the script which called it. Also, it's possible to return values from included files. You can take the value of the include call as you would for a normal function. "

Reply:
@eldy is it ok if we test on $res === TRUE ?

Reply:
i made another commit to test on TRUE and also updated the PR description to explain more deeply the reasons

Reply:
regarding the php doc, i made a last change to test on FALSE which is the return value when include fails (by my tests, it does not return TRUE when ok)

Reply:
forget everything, you were absolutely right !